modules/oe_translation_block_field/src/BlockFieldProcessor.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Drupal\oe_translation_block_field;
+
+use Drupal\block_field\BlockFieldItemInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Render\Element;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\tmgmt_content\DefaultFieldProcessor;
+
+/**
+ * Field processor for the Block reference field.
+ */
+class BlockFieldProcessor extends DefaultFieldProcessor {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function extractTranslatableData(FieldItemListInterface $field) {
+    $data = [];
+    /* @var \Drupal\Core\Field\FieldItemInterface $field_item */
+    $field_definition = $field->getFieldDefinition();
+
+    foreach ($field as $delta => $field_item) {
+      $property = $field_item->getProperties()['settings'];
+      $data['#label'] = $field_definition->getLabel();
+
+      if (count($field) > 1) {
+        // More than one item, add a label for the delta.
+        $data[$delta]['#label'] = $this->t('Delta #@delta', ['@delta' => $delta]);
```

Comment:
is it necessary don't have this with fields where we have only 1 field item?
If simplicity it also makes sense don't use `if (count($field) > 1) {`
optional comment.